test/integration/targets/firewalld/tasks/main.yml

```diff
@@ -16,15 +16,19 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-- name: Ensure firewalld is installed
-  package:
-    name: firewalld
-    state: installed
+- name: Run firewalld tests
+  block:
+    - name: Ensure firewalld is installed
+      package:
+        name: firewalld
+        state: installed
+      # This doesn't work for CentOS 6 because firewalld doesn't exist in CentOS6
 
-# Check to make sure the firewalld python module is available.
-- shell: "{{ansible_python.executable}} -c 'import firewall'"
-  register: check_output
-  ignore_errors: true
+    # Check to make sure the firewalld python module is available.
+    - shell: "{{ansible_python.executable}} -c 'import firewall'"
+      register: check_output
+      ignore_errors: true
 
-- import_tasks: test_cases.yml
-  when: check_output.rc == 0
+    - import_tasks: test_cases.yml
+      when: check_output.rc == 0
+  when: (ansible_os_family != "RedHat" and ansible_distribution_major_version != "6")
```

Comment:
Shouldn't this be `when: not (ansible_os_family == "RedHat" and ansible_distribution_major_version == "6")`

Reply:
@mattclay it 100% should, I didn't see this comment before so mine is slightly different but it's basically the same thing :+1: